### DISCUSSION TITLE
Route metadata in custom proxy middleware

### DISCUSSION THREAD

**Discussion by mrob1600** [dotnet/yarp#1169] [Q&A] [answered]
The ReverseProxy.Code.Sample has the below code in it's custom middleware. Based on the Todo comment, is there an example somewhere of looking up or using metadata in the middleware? This code sample is not exactly what I'm trying to do. I need to determine whether or not to proxy the request based on metadata of the selected route. 

It looks like ReverseProxyFeature.Route will work to get the Metadata of the route. 

```c#
// Filter destinations based on criteria
            foreach (var d in availableDestinationsFeature.AvailableDestinations)
            {
                //Todo: Replace with a lookup of metadata - but not currently exposed correctly here
                if (d.DestinationId.Contains("debug") == useDebugDestinations) { filteredDestinations.Add(d); }
            }
```

**Comment by Tratcher** [comment, marked as answer]
You're looking for IReverseProxyFeature.Route.Config.Metadata. availableDestinationsFeature in that sample is the IReverseProxyFeature, it just has an outdated name.
